controllers/repository_controller.go

```diff
@@ -108,10 +108,13 @@ func (r *RepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	_ctx, _cancel := context.WithCancel(ctx)
 	r.C[key] = repository.NewChartmuseum(_ctx, logger, r.Client, r.Scheme, repo, _cancel)
-	r.C[key].Start()
+	if err := r.C[key].Start(); err != nil {
+		r.Recorder.Event(repo, v1.EventTypeNormal, "Start", fmt.Sprintf("start %s fail", key))
```

Comment:
Here maybe you should use `EventTypeWarning`. Please @0xff-dev double confirms

Reply:
`EventTypeWarning`

Reply:
Another option is to return a list of events(or one event?), with each step returning a corresponding event if something goes wrong, as exemplified in kubernetes: https://github.com/kubernetes/kubernetes/blob/2c6c4566eff972d6c1320b5f8ad795f88c822d09/staging/src/k8s.io/endpointslice/topologycache/topologycache.go#L110-L168

Reply:
> The Reason should be words like `StartFail` or something?

Agree.

Reply:
I think reason just describes the cause of the event, which can be expressed in more detail through message.

Reply:
> Choose a unique, specific, short, CamelCase reason for each event category. For example, `FreeDiskSpaceInvalid` is a good event reason because it is likely to refer to just one situation, but `Started` is not a good reason because it doesn't sufficiently indicate what started, even when combined with other event fields.

source: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events

Reply:
how about "StartupFailed" as a reason?